### ISSUE TITLE
a11y: Checkbox ignores hideLabel prop

### ISSUE THREAD

**Issue by scottdickerson** [carbon-design-system/carbon-components-react#649]
## Detailed description

The checkbox component should support the same hideLabel prop for accessibility that TextInput does and should have labelText marked as a required property.

> Is this issue related to a specific component?
Checkbox

> What did you expect to happen? What happened instead? What would you like to see changed?
I expected behavior like the TextInput property where the label is actually present but visually hidden to support screenreaders

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
5.23.0

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

## Steps to reproduce the issue

See this code sandbox: 
https://codesandbox.io/s/71l2467kk0

## Add labels
a11y enhancement

**Comment by scottdickerson** [comment]
InlineCheckbox would work fine for my usecase and satisfies the the A11y requirement here: 
https://www.w3.org/WAI/tutorials/forms/labels/ 

I can switch to that component, thanks!

**Comment by scottdickerson** [comment]
one small problem I've had @joshblack using InlineCheckbox, is that it doesn't support the defaultChecked prop and has different callback behavior than Checkbox.  Could we not just apply the InlineCheckbox rendering logic to the Checkbox component with a new "inline" and "hideLabel" property?  This would be more inline with how Select works. http://react.carbondesignsystem.com/?selectedKind=Select&selectedStory=inline&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

**Comment by joshblack** [comment]
@scottdickerson I definitely think we can! I just wanted to verify that it met the requirements, thanks for bringing that up 😄

**Comment by scottdickerson** [comment]
After looking more closely at the InlineCheckbox implementation I'm not sure that it will work in a screenreader as the aria-label is attached to the <label> element and not the base <input element as the specification recommends.  Perhaps we can replace it with the normal Checkbox using the hideLabel prop.  Here's my PR with an implementation of hideLabel for Checkbox: https://github.com/carbon-design-system/carbon-components-react/pull/653
